spec/classes/puppet_agent_install_spec.rb

```diff
@@ -15,8 +15,14 @@
         :puppetversion => Puppet.version,
       } end
 
-      it 'should not define provider' do
```

Comment:
I prefer the pattern of the other files you changed: set a package_name variable and don't duplicate the tests.